src/measure_theory/measure_space.lean

```diff
@@ -1916,6 +2011,127 @@ end measure_space
+    exists_is_measurable_superset_of_measure_eq_zero this,
+  let g : α → β := t.piecewise (hν.mk f) (hμ.mk f),
+  refine ⟨g, measurable.piecewise t_meas hν.measurable_mk hμ.measurable_mk, _⟩,
+  change μ {x | f x ≠ g x} + ν {x | f x ≠ g x} = 0,
+  suffices : μ {x | f x ≠ g x} = 0 ∧ ν {x | f x ≠ g x} = 0, by simp [this.1, this.2],
+  have ht : {x | f x ≠ g x} ⊆ t,
+  { assume x hx,
+    by_contra h,
+    simp only [g, h, mem_set_of_eq, ne.def, not_false_iff, piecewise_eq_of_not_mem] at hx,
+    exact h (st hx) },
+  split,
+  { have : μ {x | f x ≠ g x} ≤ μ t := measure_mono ht,
+    rw μt at this,
+    exact le_antisymm this bot_le },
+  { have : {x | f x ≠ g x} ⊆ {x | f x ≠ hν.mk f x},
+    { assume x hx,
+      simpa [ht hx, g] using hx },
+    apply le_antisymm _ bot_le,
+    calc ν {x | f x ≠ g x} ≤ ν {x | f x ≠ hν.mk f x} : measure_mono this
+    ... = 0 : hν.ae_eq_mk }
+end
+
+lemma smul_measure (h : ae_measurable f μ) (c : ennreal) :
+  ae_measurable f (c • μ) :=
+⟨h.mk f, h.measurable_mk, ae_smul_measure h.ae_eq_mk c⟩
+
+lemma comp_measurable [measurable_space δ] {f : α → δ} {g : δ → β}
+  (hg : ae_measurable g (measure.map f μ)) (hf : measurable f) : ae_measurable (g ∘ f) μ :=
+⟨(hg.mk g) ∘ f, hg.measurable_mk.comp hf, ae_eq_comp hf hg.ae_eq_mk⟩
+
+lemma prod_mk {γ : Type*} [measurable_space γ] {f : α → β} {g : α → γ}
+  (hf : ae_measurable f μ) (hg : ae_measurable g μ) : ae_measurable (λ x, (f x, g x)) μ :=
+⟨λ a, (hf.mk f a, hg.mk g a), hf.measurable_mk.prod_mk hg.measurable_mk,
+  eventually_eq.prod_mk hf.ae_eq_mk hg.ae_eq_mk⟩
+
+lemma is_null_measurable (h : ae_measurable f μ) {s : set β} (hs : is_measurable s) :
+  is_null_measurable μ (f ⁻¹' s) :=
+begin
+  apply is_null_measurable_iff_ae.2,
+  refine ⟨(h.mk f) ⁻¹' s, h.measurable_mk hs, _⟩,
+  filter_upwards [h.ae_eq_mk],
+  assume x hx,
+  change (f x ∈ s) = ((h.mk f) x ∈ s),
+  rwa hx
+end
+
+end ae_measurable
+
+lemma ae_measurable_congr (h : f =ᵐ[μ] g) :
+  ae_measurable f μ ↔ ae_measurable g μ :=
+⟨λ hf, ae_measurable.congr hf h, λ hg, ae_measurable.congr hg h.symm⟩
+
+@[simp] lemma ae_measurable_add_measure_iff :
+  ae_measurable f (μ + ν) ↔ ae_measurable f μ ∧ ae_measurable f ν :=
+⟨λ h, ⟨h.mono_measure (measure.le_add_right (le_refl _)),
+         h.mono_measure (measure.le_add_left (le_refl _))⟩,
+  λ h, h.1.add_measure h.2⟩
+
+@[simp] lemma ae_measurable_const {b : β} : ae_measurable (λ a : α, b) μ :=
+measurable_const.ae_measurable
+
+@[simp] lemma ae_measurable_smul_measure_iff {c : ennreal} (hc : c ≠ 0) :
+  ae_measurable f (c • μ) ↔ ae_measurable f μ :=
+⟨λ h, ⟨h.mk f, h.measurable_mk, (ae_smul_measure_iff hc).1 h.ae_eq_mk⟩,
+  λ h, ⟨h.mk f, h.measurable_mk, (ae_smul_measure_iff hc).2 h.ae_eq_mk⟩⟩
+
+lemma measurable.comp_ae_measurable [measurable_space δ] {f : α → δ} {g : δ → β}
+  (hg : measurable g) (hf : ae_measurable f μ) : ae_measurable (g ∘ f) μ :=
+⟨g ∘ hf.mk f, hg.comp hf.measurable_mk, eventually_eq.fun_comp hf.ae_eq_mk _⟩
```

Comment:
Could you please add `measurable.ae_measurable`?

Reply:
It is already there, line 2033